openhtf/io/http_api.py

```diff
@@ -76,14 +116,93 @@ def do_POST(self):  # pylint: disable=invalid-name
           uuid.UUID((data['id'])), data['response'])
 
   def Start(self):
-    """Give the server a style-conformant Start method."""
+    """Style-compliant start method."""
     self.start()
 
   def Stop(self):
-    """Stop the server."""
+    """Stop the HTTP server."""
     self.Kill()
 
   def _ThreadProc(self):
     """Start up a raw HTTPServer based on our HTTPHandler definition."""
     server = BaseHTTPServer.HTTPServer(('', FLAGS.http_port), self.HTTPHandler)
     server.serve_forever()
+
+
+class MulticastAgent(util.threads.KillableThread):
```

Comment:
There is shared data as well (the string used for pinging). I could be convinced either way on this one. Putting the Ping method into the server class was an arbitrary decision on my part.